Source/Scene/ShadowMap.js

```diff
@@ -1211,16 +1233,90 @@ define([
         }
     }
 
+    var scratchCartesian1 = new Cartesian3();
+    var scratchCartesian2 = new Cartesian3();
+    var scratchBoundingSphere = new BoundingSphere();
+    var scratchCenter = scratchBoundingSphere.center;
+
+    function checkVisibility(shadowMap, frameState) {
+        var camera = frameState.camera;
+        var lightCamera = shadowMap._lightCamera;
+        var sceneCamera = shadowMap._sceneCamera;
+        var shadowMapCamera = shadowMap._shadowMapCamera;
+
+        var boundingSphere = scratchBoundingSphere;
+
+        // Check whether the shadow map is in view and needs to be updated
+        if (shadowMap._cascadesEnabled) {
+            // If the nearest shadow receiver is further than the shadow map's maximum distance then the shadow map is out of view.
+            if (sceneCamera.frustum.near > shadowMap._maximumDistance) {
+                shadowMap._outOfView = true;
+                shadowMap._needsUpdate = false;
+                return;
+            }
+
+            // If the light source is below the horizon then the shadow map is out of view
+            var surfaceNormal = frameState.mapProjection.ellipsoid.geodeticSurfaceNormal(sceneCamera.positionWC, scratchCartesian1);
+            var lightDirection = Cartesian3.negate(shadowMapCamera.directionWC, scratchCartesian2);
+            var dot = Cartesian3.dot(surfaceNormal, lightDirection);
+            if (dot < 0.1) {
```

Comment:
Using the Sandcastle I just cut if off at a light horizon above 85 degrees. Maybe that's too much.